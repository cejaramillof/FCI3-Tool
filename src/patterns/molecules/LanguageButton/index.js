import React from 'react';
import Button from 'patterns/atoms/Button';


const LanguageButton = props => {
    const cssLanguage = {
        en: (props.lang === 'en') ? 'active en-active': '',
        es: (props.lang === 'es') ? 'active es-active': ''
    }
    
    return (
        <Button
            type='button'
            className={`btn btn-secondary ${props.currentLang === 'en' ? cssLanguage.en : cssLanguage.es}`} 
            disabled={!props.lang} 
            onClick={()=>{props.changeLanguage(props.lang)}}
        >
            {props.lang}
        </Button>
    )
}

export default LanguageButton;