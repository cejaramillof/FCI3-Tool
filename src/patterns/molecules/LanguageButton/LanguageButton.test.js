import React from 'react';
import { mount } from 'enzyme';
import sinon from 'sinon';
import LanguageButton from './index';

describe('LanguageButton test in English mode', () => {
    const props = {
        lang: 'en',
        onClick: sinon.spy(),
        currentLang: 'en'
    }
    
    const wrapper = mount(
      <LanguageButton {...props} />
    );
    
    it('Should render correctly', () => expect(wrapper).toMatchSnapshot());
    
    it('Should render with English lang', () => expect(wrapper.props().lang).toEqual('en'));
    
    it('Should render with currentLang', () => expect(
        wrapper.props().lang).toEqual('en')
    );
});

describe('LanguageButton test in Espanish mode', () => {
    const props = {
        lang: 'es',
        onClick: sinon.spy(),
        currentLang: 'es'
    }
    
    const wrapper = mount(
      <LanguageButton {...props} />
    );
    
    it('Should render correctly', () => expect(wrapper).toMatchSnapshot());
    
    it('Should render with Espanish lang', () => expect(wrapper.props().lang).toEqual('es'));
    
    it('Should render with currentLang', () => expect(
        wrapper.props().lang).toEqual('es')
    );
})







