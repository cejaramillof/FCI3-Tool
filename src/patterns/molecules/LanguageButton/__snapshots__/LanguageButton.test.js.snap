// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageButton test in English mode Should render correctly 1`] = `ReactWrapper {}`;

exports[`LanguageButton test in Espanish mode Should render correctly 1`] = `ReactWrapper {}`;
